src/MemoryTracing/MemoryInfoProducer.cpp

```diff
@@ -36,24 +40,66 @@ void MemoryInfoProducer::SetExitRequested(bool exit_requested) {
   exit_requested_ = exit_requested;
 }
 
-void MemoryInfoProducer::Run() {
-  orbit_base::SetCurrentThreadName("MemInfoPr::Run");
+MemoryInfoProducerRunFn SystemMemoryInfoProducerRunFn() {
+  return [](uint64_t sampling_period_ns, MemoryInfoListener* listener, int32_t /*pid*/,
+            bool* exit_requested, absl::Mutex* exit_requested_mutex) {
```

Comment:
```suggestion
void SystemMemoryInfoProducerRunFn(uint64_t sampling_period_ns, MemoryInfoListener* listener, int32_t /*pid*/,
            bool* exit_requested, absl::Mutex* exit_requested_mutex) {
```

no need to return a lambda here the function itself should work.

Reply:
Changed accordingly. Thanks!